vendors/vendor/boards/board/ports/ota_pal_for_aws/ota_pal.h

```diff
@@ -0,0 +1,231 @@
+ *
+ * @note The input OtaFileContext_t pFileContext is checked for NULL by the OTA agent before this
+ * function is called.
+ *
+ * This function may be called before the file is opened, so the file pointer pFileContext->fileHandle
+ * may be NULL when this function is called.
+ *
+ * The file pointer will be set to NULL after this function returns.
+ *
+ * @param[in] pFileContext OTA file context information.
+ *
+ * @return The OtaPalStatus_t error code is a combination of the main OTA PAL interface error and
+ *         the MCU specific sub error code. See ota_platform_interface.h for the OtaPalMainStatus_t
+ *         error codes and your specific PAL implementation for the sub error code.
+ *
+ * Major error codes returned are:
+ *
+ *   OtaPalSuccess: Aborting access to the open file was successful.
+ *   OtaPalFileAbort: Aborting access to the open file context was unsuccessful.
+ */
+OtaPalStatus_t otaPal_Abort( OtaFileContext_t * const pFileContext );
+
+/**
+ * @brief Create a new receive file for the data chunks as they come in.
+ *
+ * @note Opens the file indicated in the OTA file context in the MCU file system.
+ *
+ * @note The previous image may be present in the designated image download partition or file, so the
+ * partition or file must be completely erased or overwritten in this routine.
+ *
+ * @note The input OtaFileContext_t pFileContext is checked for NULL by the OTA agent before this
+ * function is called.
+ * The device file path is a required field in the OTA job document, so pFileContext->pFilePath is
+ * checked for NULL by the OTA agent before this function is called.
+ *
+ * @param[in] pFileContext OTA file context information.
+ *
+ * @return The OtaPalStatus_t error code is a combination of the main OTA PAL interface error and
+ *         the MCU specific sub error code. See ota_platform_interface.h for the OtaPalMainStatus_t
+ *         error codes and your specific PAL implementation for the sub error code.
+ *
+ * Major error codes returned are:
+ *
+ *   OtaPalSuccess: File creation was successful.
+ *   OtaPalRxFileTooLarge: The OTA receive file is too big for the platform to support.
+ *   OtaPalBootInfoCreateFailed: The bootloader information file creation failed.
+ *   OtaPalRxFileCreateFailed: Returned for other errors creating the file in the device's non-volatile memory.
+ */
+OtaPalStatus_t otaPal_CreateFileForRx( OtaFileContext_t * const pFileContext );
+
+/**
+ * @brief Authenticate and close the underlying receive file in the specified OTA context.
+ *
+ * @note The input OtaFileContext_t pFileContext is checked for NULL by the OTA agent before this
+ * function is called. This function is called only at the end of block ingestion.
+ * otaPAL_CreateFileForRx() must succeed before this function is reached, so
+ * pFileContext->fileHandle(or pFileContext->pFile) is never NULL.
+ * The certificate path on the device is a required job document field in the OTA Agent,
```

Comment:
This is optional field so should be checked if it is valid cert path or tag/name